spinalcordtoolbox/scripts/sct_check_dependencies.py

```diff
@@ -73,7 +73,9 @@ def resolve_module(framework_name):
         # Importing `matplotlib_inline` requires IPython, but we don't install IPython (on purpose). This is because
         # `matplotlib_inline` is only needed to run SCT scripts in Jupyter notebooks, and IPython would already be
         # installed in the parent context. So, we map `matplotlib-inline` to None to skip import (which would fail).
-        'matplotlib-inline': (None, False)
+        'matplotlib-inline': (None, False),
+        # We specify the version of `onnx` on behalf of ivadomed -- we don't import it directly. So, skip checking its version.
+        'onnx': (None, False),
```

Comment:
It looks like `onnx==1.17.0` only works if we explicitly avoid importing `onnx` during `sct_check_dependencies` (see this skip here).

If we want `onnx` to work universally, we'd have to downgrade even further to `onnx==1.16.1` and below.